lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -2786,3 +2786,128 @@ def test_group_access(self, role_name, course_is_cohorted, thread_group_state):
             self.assertFalse(expected_error)
         except Http404:
             self.assertTrue(expected_error)
+
+
+@ddt.ddt
+class RetrieveThreadTest(
```

Comment:
One additional test that would be good to have (or refactor from another test class for this Retrieve method):
  test access to cohorted threads.
    - if a thread is assigned to a cohort, only users in that cohort are allowed to access to that thread (unless they have staff/admin access).